src/renderers/WebGLRenderer.js

```diff
@@ -1990,6 +1990,10 @@ function WebGLRenderer( parameters ) {
 				m_uniforms.color.value.copy( material.color );
 				m_uniforms.opacity.value = material.opacity;
 
+			} else if ( material.envMap ) {
+
+				refreshUniformsCommon( m_uniforms, material );
```

Comment:
It seems this new line has broken [webgl_materials_nodes](https://raw.githack.com/mrdoob/three.js/dev/examples/webgl_materials_nodes.html). When this line is called with `MeshStandardNodeMaterial`, a runtime error occurs since the respective `m_uniforms` variable does not have a `opacity` uniform.

Reply:
I think that be good if we can follow the same standard of check class pattern material.isMeshPhongMaterial, material.isMeshStandardMaterial...

Reply:
Ah, sorry about that! I'm not familiar with the nodes system yet. @sunag, are you working on a fix for this, or are you asking me to put up a PR?

Reply:
No specifically, but at least related, is a redesign of uniforms updates ( https://github.com/mrdoob/three.js/pull/18068 ), but it may take too long... 

I like of the simplification of `boxMesh.material.uniforms.tCube.value` to `boxMesh.material.envMap`, the problem is that `ShaderMaterial` use property like `envMap` input to an independent use like `Node` in `NodeMaterial` for example. I think that at the moment is more easy revert the changes of `material.envMap` to `boxMesh.material.uniforms.tCube.value`, otherwise to this work properly is necessary a materialClass to justify a flag specific like `material.isEnvMapMaterial`.

Reply:
>I like of the simplification of boxMesh.material.uniforms.tCube.value to boxMesh.material.envMap

I do not like it. The box mesh does not have an environment map -- it has a texture. That change has conflated concepts. If it is reverted, that is fine with me.

Reply:
Justifying a little more... The maily problem that I see is that the `material.envMap` without class approach is a hack for any material including `RawShader` aparently.

Reply:
Fair enough, but what kind of texture would you put on a skybox besides an environment map? As it was we had a bunch of texture types specific to environment maps that were not usable as skyboxes, so we ended up needing two different textures in memory representing the same image just to have a matching environment and skybox. I'm sure there's a better and clearer way then exactly what I've done, but I'm hoping we can at least find a method that allows for reuse.

Reply:
@elalish Agreed, your argument has merits. There are downsides, however, as noted.

Also in the cube vertex shader, we now have `vReflect = vWorldDirection`, which is not a reflection vector.

I do not know what the best solution is at the moment, but we may get to it by figuring out how to fix Node.

Reply:
I'll take care of this.

Reply:
I think this should solve the issue:
https://github.com/mrdoob/three.js/commit/b1d1f26b19d78b0d792bb86305680d82b2fc9005